### PR TITLE
Feature/settings

### PR DESCRIPTION
**Pull request by Sorcere666**
-Please don't explode on me. QQ.

### CODE REVIEW

File: app/src/main/java/org/sil/gatherwords/AppCompatPreferenceActivity.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import android.support.v7.widget.Toolbar;
+import android.view.MenuInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+/**
+ * A {@link android.preference.PreferenceActivity} which implements and proxies the necessary calls
+ * to be used with AppCompat.
+ */
+public abstract class AppCompatPreferenceActivity extends PreferenceActivity {
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/AppCompatPreferenceActivity.java]: What is the purpose of this wrapper?  It does not appear to be in use.

File: app/src/main/java/org/sil/gatherwords/SettingsActivityCustom.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package org.sil.gatherwords;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+
+public class SettingsActivityCustom extends AppCompatActivity {
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/SettingsActivityCustom.java]: Could you rename this?  `XXCustom` gives very little idea of what this class is relevant to.

File: app/src/main/java/org/sil/gatherwords/MainActivity.java [hunk 3]
```diff
@@ -1,14 +1,78 @@
...
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+
+        switch(item.getItemId()){
+
+            //Links to the preferences/settings page.
+            case R.id.app_action_bar_settings:
+
+                Toast.makeText(getApplicationContext(), "Settings", Toast.LENGTH_SHORT).show();
```
Comment (Andrewdt97) [inline app/src/main/java/org/sil/gatherwords/MainActivity.java]: Delete this toast. It's not really necessary and a bit distracting.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,5 +1,55 @@
...
+    <!-- =============================================================== -->
+    <!-- =============================================================== -->
+
+
+
+    <!-- Strings for the language selection list-type preference setting -->
+    <!-- =============================================================== -->
+    <!-- =============================================================== -->
+    <string name="pref_language_select_key">list_pref_1anguage</string>
+    <string name="pref_language_select_title">Language Selection.</string>
```
Comment (Andrewdt97) [inline app/src/main/res/values/strings.xml]: `<string name="pref_language_select_key">list_pref_1anguage</string>
 <string name="pref_language_select_title">Language Selection.</string>`

Delete the periods from the end of these strings

File: app/src/main/java/org/sil/gatherwords/SettingsWordFieldSetupFragment.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mParam1 = getArguments().getString(ARG_PARAM1);
+            mParam2 = getArguments().getString(ARG_PARAM2);
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/SettingsWordFieldSetupFragment.java]: What is the purpose of these parameters?  You store and load them, but I do not see them used.

File: app/src/main/java/org/sil/gatherwords/SettingsWordFieldSetupFragment.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+        if (mListener != null) {
+            mListener.onFragmentInteraction(uri);
+        }
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if (context instanceof OnFragmentInteractionListener) {
+            mListener = (OnFragmentInteractionListener) context;
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/SettingsWordFieldSetupFragment.java]: It does not appear you are using listeners, so it should be safe to delete `onButtonPressed()`, `onAttach()`, `onDetach()`, and `OnFragmentInteractionListener()`.

File: app/src/main/java/org/sil/gatherwords/MainActivity.java [hunk 7]
```diff
@@ -17,6 +23,27 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        /* Begin code by Joseph Jinn */
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/MainActivity.java]: In most cases, we do not label our code like this (includes the other instances).  If it becomes necessary to determine the author, we can use git history.

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1,6 +1,83 @@
...
+    <string name="pref_language_option_title">Language Options.</string>
+    <string name="pref_language_option_summary">Language and associated options.</string>
+    <string-array name="pref_language_option_default_values">
+    </string-array>
+    <string-array name="pref_language_option_entries">
+        <item>English</item>
+        <item>Thai</item>
+        <item>Latin</item>
+    </string-array>
+    <string-array name="pref_language_option_values">
```
Comment (alpha0010) [inline app/src/main/res/values/strings.xml]: Did you mean `integer-array`?

**Pull request closed without merging**